### PR TITLE
Do not aggregate header info over multiple calls

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR
@yngveaasheim FYI
This plugs a huge memory leak.

### CODE REVIEW

**Review by havardpe — COMMENTED** [review]
what about headerinfoPos?

File: fbench/src/httpclient/httpclient.h [hunk 1]
```diff
@@ -95,21 +95,20 @@ class HTTPClient
...
-  uint64_t         _reuseCount;
+  std::string          _host_header_value;
+  uint64_t             _reuseCount;
 
   size_t           _bufsize;
   char            *_buf;
   ssize_t          _bufused;
   ssize_t          _bufpos;
 
-  std::string      _headerinfo;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply fbench/src/httpclient/httpclient.h]: I think it is better to have the compiler enforce actual life being the same as semantic lifetime.

File: fbench/src/httpclient/httpclient.h [hunk 2]
```diff
@@ -95,21 +95,20 @@ class HTTPClient
...
   unsigned int     _headerinfoPos;
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply fbench/src/httpclient/httpclient.h]: Not used, removed.

File: fbench/src/httpclient/httpclient.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (baldersheim) [reply fbench/src/httpclient/httpclient.h]: The methods only had internal usage, so I made them private. The the API clarity is not much of a problem. Ideally it should have returned an object.

**Pull request merged**